analyses/cell-type-ewings/template_notebooks/02-cellassign.Rmd

```diff
@@ -9,11 +9,11 @@ output:
 params:
   sample_id: SCPCS000490
   library_id: SCPCL000822
-  marker_gene_classification: "../results/annotate_tumor_cells_output/SCPCS000490/SCPCL000822_tumor-normal-classifications.tsv"
-  tumor_marker_predictions: "../results/annotate_tumor_cells_output/SCPCS000490/cellassign/SCPCL000822_tumor-only-predictions.tsv"
-  visser_marker_predictions: "../results/annotate_tumor_cells_output/SCPCS000490/cellassign/SCPCL000822_visser-predictions.tsv"
-  panglao_predictions: "../results/annotate_tumor_cells_output/SCPCS000490/cellassign/SCPCL000822_panglao-predictions.tsv"
-  results_dir: "../results/annotate_tumor_cells_output/SCPCS000490"
+  marker_gene_classification: "../results/sim_annotate_tumor/SCPCS000490/SCPCL000822_tumor-normal-classifications.tsv"
+  tumor_marker_predictions: "../results/sim_annotate_tumor/SCPCS000490/cellassign/SCPCL000822_tumor-marker_predictions.tsv"
+  visser_marker_predictions: "../results/sim_annotate_tumor/SCPCS000490/cellassign/SCPCL000822_visser-all-marker_predictions.tsv"
+  panglao_predictions: "../results/sim_annotate_tumor/SCPCS000490/cellassign/SCPCL000822_panglao-endo-fibro_predictions.tsv"
+  results_dir: "../results/sim_annotate_tumor/SCPCS000490"
```

Comment:
I feel like you probably want to leave these defaults as the full output directory and change them in the rendering script itself as needed?

I also expect you may not want all of these defined as `params`, but you might instead want to build these paths within the notebook from `sample_id`, `library_id` and `results_dir`? I see from the outer script why you might want to pass these in, but in that case I think it might make sense to leave the defaults as generic names, rather than explicit paths.

For the outer script, I might add a variable that designates if this is a test run, so you can change the path for test data on the fly without having to modify any other defaults.

Reply:
The reason I have full paths here is because the predictions files have to be defined in the outer script since they are used both to run cell assign and to render the notebook. I thought it would be easier to maintain if the full paths to these files were defined only in the outer script and not in both the script and the notebook. 

I can adjust the defaults to be just generic file names, this is mostly left over from me testing things and trying to figure out why something was failing. 

> For the outer script, I might add a variable that designates if this is a test run, so you can change the path for test data on the fly without having to modify any other defaults.

I'm not totally sure what you mean here? You would just change the `data_dir` argument if you wanted to explicitly specify test vs real data. I don't think adding another argument really helps here.

Reply:
Probably not, but you would want to change both `data_dir` and `workflow_results_dir`, right? Or really just `workflow_results_dir` as the download script should handle switching the input directories?

https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/8937a0aed3e36837b2f785e85dbf38a9dbe705a2/analyses/cell-type-ewings/annotate-tumor-cells.sh#L47C1-L48C104

Reply:
You would only need to change `data_dir` if you want to switch back and forth between what's considered "current". So if you have both test data and real data downloaded, then you would want to specify which input to use. Otherwise it's directed to whatever you downloaded more recently. For the purposes of GHA, it's only downloading the test data, so the "current" directory will automatically link to the test data. 

You would only need to change the `workflow_results_dir` if you want to save the output to a different folder. This would also only need to happen if you are working locally with both test and regular data from the same samples. In the GHA, there is only one output so this doesn't need to be changed. I guess you could create a test flag rather than changing the `workflow_results_dir`, but I thought we were keeping the number of arguments here to a minimum?